master/internal/configpolicy/task_config_policy.go

```diff
@@ -160,6 +163,53 @@ func GetMergedConstraints(ctx context.Context, workspaceID int, workloadType str
 	return &constraints, nil
 }
 
+// MergeWithInvariantExperimentConfigs merges the config with workspace and global invariant
+// configs, where a global invariant config takes precedence over a workspace-level invariant
+// config.
+func MergeWithInvariantExperimentConfigs(ctx context.Context, workspaceID int,
+	config *expconf.ExperimentConfigV0,
+) error {
+	originalConfig := *config
+	var wkspOverride, globalOverride bool
+	wkspConfigPolicies, err := GetTaskConfigPolicies(ctx, &workspaceID, model.ExperimentType)
+	if err != nil {
+		return err
+	}
+	if wkspConfigPolicies.InvariantConfig != nil {
+		if err := json.Unmarshal([]byte(*wkspConfigPolicies.InvariantConfig), config); err != nil {
+			return fmt.Errorf("error unmarshaling workspace invariant config: %w", err)
+		}
+		wkspOverride = true
+	}
+
+	globalConfigPolicies, err := GetTaskConfigPolicies(ctx, nil, model.ExperimentType)
+	if err != nil {
+		return err
+	}
+	if globalConfigPolicies.InvariantConfig != nil {
+		if err = json.Unmarshal([]byte(*globalConfigPolicies.InvariantConfig), config); err != nil {
+			return fmt.Errorf(" error unmarshaling global invariant config: %w", err)
+		}
```

Comment:
Do we want to set `globalOverride = true` here?

Reply:
Good catch! Added